mono/metadata/threads.c

```diff
@@ -2009,6 +2009,8 @@ ves_icall_System_Threading_WaitHandle_Wait_internal (gpointer *handles, gint32 n
 	return map_native_wait_result_to_managed (ret, numhandles);
 }
 
+#if G_HAVE_API_SUPPORT(HAVE_CLASSIC_WINAPI_SUPPORT)
```

Comment:
Where is it defined on non-classic API? Should we also put the declaration in `HAVE_API_SUPPORT(HAVE_CLASSIC_WINAPI_SUPPORT)`?

Reply:
It is not defined. The reason for it is that it wouldn't compile as mono_win32_signal_object_and_wait isn't defined either.

Reply:
SignalObjectAndWait is available for UWP.
I believe people use it to implement slow condition variables (should use the Vista ones instead -- unless you need more likely to be reentrant then stick to kernel calls, though the use of GetLastError makes the kernel wrappers still not reentrant, a peeve of mine).

https://msdn.microsoft.com/en-us/library/windows/desktop/ms686293(v=vs.85).aspx

Reply:
Again, this only got added in a recent SDK update. We still need to support older SDKs for now.

Reply:
Can the code adapt to newer SDKs automatically?

Reply:
Or use older SDK and bind dynamically?

Reply:
LoadLibrary is not allowed on UWP. And we still need to support older SDKs for now...

Reply:
@luhenry @jaykrell Is this code fine to take as-is?

Reply:
I think so.

Reply:
:+1: